.github/workflows/coverage-check.yml

```diff
@@ -0,0 +1,116 @@
+name: Check Test Coverage
+
+on:
+  pull_request:
+    branches: [development]
+
+jobs:
+  test-coverage:
+    name: Check Test Coverage
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+
+    steps:
+      - name: Checkout development branch
+        uses: actions/checkout@v3
+        with:
+          ref: development
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: "18"
+
+      - name: Install dependencies
+        run: sudo apt -y update && sudo apt -y install libusb-1.0-0-dev libudev-dev
+
+      - name: Yarn Install
+        run: yarn install --mode=skip-build && yarn allow-scripts
+
+      - name: Setup Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Setup Forge
+        run: yarn workspace @ubiquity/contracts forge:install
+
+      - name: Get development branch coverage
+        run: |
+          # this is temporary, until we merge the pr branch into development then we can use the yarn workspaces command
+
+          mkdir packages/contracts/scripts/coverage-check && touch packages/contracts/scripts/coverage-check/coverage.ts
+
+          echo 'import { getRPC } from "../runner/rpcutil";
+          import { exec } from "child_process";
+          const setRPCURL = async () => {
+            const optimalRPC = await getRPC();
+            console.log(optimalRPC);
+            return optimalRPC;
+          };
+
+          setRPCURL();' > packages/contracts/scripts/coverage-check/coverage-check.ts    
+
+          RPC_URL=$(npx ts-node packages/contracts/scripts/coverage-check/coverage-check.ts)
+          echo $RPC_URL
+          echo "forge coverage --no-match-path=**/utils/**=**/utils/** --rpc-url $RPC_URL"
```

Comment:
@pwilson77 

why do we need this option `--no-match-path=**/utils/**=**/utils/**`?

if you want to exclude some files from the coverage report then it is better to use foundry's config

Reply:
ok will update foundry's config